src/main/java/com/example/newslettercore/infrastructure/repository/jpa/newsletter/JpaNewsletterMapper.java

```diff
@@ -0,0 +1,93 @@
+package com.example.newslettercore.infrastructure.repository.jpa.newsletter;
+
+import com.example.newslettercore.domain.newsletter.model.Newsletter;
+import com.example.newslettercore.domain.newsletter.model.Template;
+import org.apache.commons.collections4.CollectionUtils;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.factory.Mappers;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+
+@Mapper
+public interface JpaNewsletterMapper {
+
+    JpaNewsletterMapper getMapper = Mappers.getMapper(JpaNewsletterMapper.class);
+
+    NewsletterEntity mapToNewsletterEntity(Newsletter newsletter);
+
+    @AfterMapping
+    default void afterMapToNewsletterEntity(Newsletter newsletter, @MappingTarget NewsletterEntity newsletterEntity) {
+
+        if (CollectionUtils.isNotEmpty(newsletter.getTemplates())) {
+            newsletterEntity.getTemplates() //
```

Comment:
Sprawdzałeś czy jest to potrzebne? Wydaje mi się, że JPA poradzi sobię bez tego i sam ustawi referencję, jeżeli są zdefiniowane mappedBy w encjach.

Reply:
Jest potrzebne, bez tego bym mial cykliczne wywolania w funkcjach wygenerowanych przez mapper

Reply:
Hmm, chyba nie rozumiem, ale to wdzwońmy się w poniedziałek. Dam ci approva bo to mała rzecz, żebyś mógł dalej pracować nad projektem.